frontend/packages/knative-plugin/src/components/test-function/modal-components/ModalCompact.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { FormikProps, FormikValues } from 'formik/dist/types';
+import { HTTPMessageTypes, RequestPaneTab, ResponsePaneTab } from '../../../types';
+import RequestBody from './RequestBody';
+import RequestOptions from './RequestOptions';
+import ResponseBody from './ResponseBody';
+import ResponseInfo from './ResponseInfo';
+import '../TestFunctionModal.scss';
+
+const ModalCompact: React.FC<FormikProps<FormikValues>> = (props) => {
+  const [activeParentTab, setActiveParentTab] = React.useState<HTTPMessageTypes>(
+    HTTPMessageTypes.Request,
+  );
+  const [activeTab, setActiveTab] = React.useState<RequestPaneTab | ResponsePaneTab>(
+    RequestPaneTab.Body || ResponsePaneTab.Body,
+  );
+  return (
+    <>
+      <div className="co-m-horizontal-nav">
+        <ul className="co-m-horizontal-nav__menu">
```

Comment:
We should check if we can implement this with PF components. And we should check if we can use a Tab component instead of a NavBar here. But we can keep that until we have finalized the design and change it with the 2nd PR if needed.

https://www.patternfly.org/v4/components/navigation
https://www.patternfly.org/v4/components/tabs/